File: app/src/main/java/info/staticfree/SuperGenPass/PasswordGenerationException.java
package info.staticfree.SuperGenPass;

import android.support.annotation.NonNull;

/**
 * An exception raised if there was a problem generating a password with the given criteria.
 */
public class PasswordGenerationException extends Exception {

    public PasswordGenerationException(@NonNull String string,
            @NonNull Throwable source) {
        super(string, source);
    }

    public PasswordGenerationException(String string) {
        super(string);
    }

    private static final long serialVersionUID = 6491091736643793303L;
}

File: app/src/main/java/info/staticfree/supergenpass/PasswordGenerationException.kt
package info.staticfree.supergenpass

/**
 * An exception raised if there was a problem generating a password with the given criteria.
 */
open class PasswordGenerationException : Exception {
    constructor(
        string: String,
        source: Throwable
    ) : super(string, source) {
    }

    constructor(string: String?) : super(string)

    companion object {
        private const val serialVersionUID = 6491091736643793303L
    }
}